packages/xo-web/src/xo-app/vm/tab-advanced.js

```diff
@@ -306,13 +316,41 @@ const NIC_TYPE_OPTIONS = [
 export default class TabAdvanced extends Component {
   componentDidMount () {
     getVmsHaValues().then(vmsHaValues => this.setState({ vmsHaValues }))
+    this.setState({
+      cpuMask: this._getCpuMask(),
```

Comment:
You should only rely on the props, here. No need for a `state` value.

Reply:
the `_getCpuMask` get `mask` from `props`.  
I use it to formulate value of `cpuMask`  and use it for `XoSelect's value`

Reply:
But why don't you call `_getCpuMask` directly when you pass it to `XoSelect`? Why do you need to store it in the state?

Reply:
When we call `onChange`  we don't wait too much to see changes in the `input` of `Select`.

Reply:
It doesn't matter if we need to wait half a second. The important thing is to show the actual value of the CPU mask. Here, even if there is an error on the server side, you'll still show the fake new value.